src/main/java/blf/blockchains/ethereum/instructions/EthereumTransactionFilterInstruction.java

```diff
@@ -39,28 +38,33 @@ public EthereumTransactionFilterInstruction(
     }
 
     @Override
-    public void execute(ProgramState state) throws ProgramException {
+    public void execute(ProgramState state) {
         final EthereumProgramState ethereumProgramState = (EthereumProgramState) state;
         final EthereumDataReader ethereumReader = ethereumProgramState.getReader();
 
         for (EthereumTransaction tx : ethereumReader.getCurrentBlock()) {
-            if (this.senderCriterion.test(state, tx.getFrom()) && this.recipientCriterion.test(state, tx.getTo())) {
-                try {
-                    ethereumReader.setCurrentTransaction(tx);
-                    this.executeInstructions(state);
-                } catch (Exception cause) {
-                    final String message = String.format(
-                        "Error mapping transaction '%s' in block '%s'.",
-                        tx.getTransactionIndex(),
-                        tx.getBlockNumber()
-                    );
-                    final boolean abort = state.getExceptionHandler().handleExceptionAndDecideOnAbort(message, cause);
-                    if (abort) {
-                        throw new ProgramException(message, cause);
+            try {
+                if (this.senderCriterion.test(state, tx.getFrom()) && this.recipientCriterion.test(state, tx.getTo())) {
+                    try {
+                        ethereumReader.setCurrentTransaction(tx);
+                        this.executeNestedInstructions(state);
+                    } catch (Exception cause) {
+                        final String message = String.format(
+                            "Error mapping transaction '%s' in block '%s'.",
+                            tx.getTransactionIndex(),
+                            tx.getBlockNumber()
+                        );
+                        final boolean abort = state.getExceptionHandler().handleExceptionAndDecideOnAbort(message, cause);
+                        if (abort) {
```

Comment:
I don't believe this is necessary anymore. Isn't the plan to make handleExceptionAndDecideOnAbort directly abort the program and not return a boolean anymore? Also, if it returns true, you just call it again.

Reply:
You are right!